src/utils/idpLoader.ts

```diff
@@ -0,0 +1,50 @@
+import { ReadableReporter } from "italia-ts-commons/lib/reporters";
+import nodeFetch from "node-fetch";
+import { DOMParser } from "xmldom";
+import { IDPEntityDescriptor } from "../types/IDPEntityDescriptor";
+import { log } from "./logger";
+
+const EntityDescriptorTAG = "md:EntityDescriptor";
+const X509CertificateTAG = "ds:X509Certificate";
+const SingleSignOnServiceTAG = "md:SingleSignOnService";
+const SingleLogoutServiceTAG = "md:SingleLogoutService";
+
+export async function parseIdpMetadata(
+  ipdMetadataPage: string
+): Promise<ReadonlyArray<IDPEntityDescriptor>> {
+  const domParser = new DOMParser().parseFromString(ipdMetadataPage);
+  const entityDescriptors = domParser.getElementsByTagName(EntityDescriptorTAG);
+  return Array.from(entityDescriptors).reduce(
+    (idps: ReadonlyArray<IDPEntityDescriptor>, element: Element) => {
+      const certs = Array.from(
+        element.getElementsByTagName(X509CertificateTAG)
+      ).map(_ => _.textContent);
+      const elementInfoOrErrors = IDPEntityDescriptor.decode({
+        cert: certs,
+        entityID: element.getAttribute("entityID"),
+        entryPoint: (element
+          .getElementsByTagName(SingleSignOnServiceTAG)
+          .item(0) as Element).getAttribute("Location"),
+        logoutUrl: (element
+          .getElementsByTagName(SingleLogoutServiceTAG)
+          .item(0) as Element).getAttribute("Location")
+      });
+      if (elementInfoOrErrors.isLeft()) {
+        log.warning(
+          "Invalid md:EntityDescriptor. Skipping ...",
+          ReadableReporter.report(elementInfoOrErrors)
```

Comment:
the reporter will generate a multi-line report, it's better to have everything on one line when logging:

```suggestion
          ReadableReporter.report(elementInfoOrErrors).replace(/\n/g, " / ")
```

Reply:
`ReadableReporter.report(elementInfoOrErrors)` returns an array.
This code is ok?
```
ReadableReporter.report(elementInfoOrErrors)
     .reduce((report, _) => report + "\n" + _, "")
     .replace(/\n/g, " / ")
```

Reply:
fixed on 5323876

Reply:
sorry my mistake, the correct function is [errorsToReadableMessages](https://github.com/teamdigitale/io-ts-commons/blob/master/src/reporters.ts#L31):

```ts
errorsToReadableMessages(elementInfoOrErrors).join(" / ")
```

Reply:
fix on commit 6b3100d